src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1842,9 +1843,79 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
             VisitRvalue(node.AccessExpression);
             IntersectWith(ref this.State, ref receiverState);
 
+            TypeSymbol type = node.Type;
+            NullableAnnotation resultAnnotation;
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+
+            if (type.SpecialType == SpecialType.System_Void || type.IsErrorType() || _resultType.IsNull)
+            {
+                resultAnnotation = NullableAnnotation.Unknown;
+            }
+            else if (_resultType.IsPossiblyNullableReferenceTypeTypeParameter())
+            {
+                Debug.Assert(_resultType.TypeSymbol == type);
+                Conversion conversion;
+
+                if (!receiverType.GetValueNullableAnnotation().IsAnyNullable())
+                {
+                    resultAnnotation = NullableAnnotation.NotNullable; // Inherit nullability of the access
+                }
+                else if (receiverType.IsPossiblyNullableReferenceTypeTypeParameter() &&
+                    (conversion = _conversions.ClassifyConversionFromType(receiverType.TypeSymbol, _resultType.TypeSymbol, ref useSiteDiagnostics)).Exists &&
+                    !conversion.IsUserDefined)
+                {
+                    // where T : U
+                    // T ?? U or U ?? T
```

Comment:
>// T ?? U or U ?? T [](start = 20, length = 19)

Conditional access is `?.` and `?[]` right? I'm confused by the comments that mention `??`

Reply:
> Conditional access is ?. and ?[] right? I'm confused by the comments that mention ??

Yes, this is a typo. Will fix.

---
In reply to: [236875235](https://github.com/dotnet/roslyn/pull/31264#discussion_r236875235) [](ancestors = 236875235)